sources/lib/api/gcp/bigquery/_federated_table.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+"""Implemenents External Table functionality."""
+
+import gcp
+import _api
+import _csv_options
+import _job
+
+
+class FederatedTable(object):
+
+  @staticmethod
+  def from_storage(source, source_format='csv', csv_options=None, ignore_unknown_values=False,
+                   max_bad_records=0, compressed=False, schema=None):
+
+    """ Create an external table for a GCS object.
+
+    Args:
+      source: the URL of the source objects(s). Can include a wildcard '*' at the end of the item
+         name. Can be a single source or a list.
+      source_format: the format of the data, 'csv' or 'json'; default 'csv'.
+      csv_options: For CSV files, the options such as quote character and delimiter.
+      ignore_unknown_values: If True, accept rows that contain values that do not match the schema;
+          the unknown values are ignored (default False).
+      max_bad_records The maximum number of bad records that are allowed (and ignored) before
+          returning an 'invalid' error in the Job result (default 0).
+      compressed: whether the data is GZ compressed or not (default False). Note that compressed
+          data can be used as a federated table but cannot be loaded into a BQ Table.
+      schema: the schema of the data. This is required for this table to be used as a federated
+          table or to be loaded using a Table object that itself has no schema (default None).
+
+  """
+    return FederatedTable(source, source_format, csv_options, ignore_unknown_values,
+                          max_bad_records, compressed, schema)
+
+  def __init__(self, source, source_format, csv_options, ignore_unknown_values, max_bad_records,
```

Comment:
We don't need to have a ctor that takes args... it can be a parameterless ctor, and the static factory method can set the various attributes.

My question is this - when there is another source, what will the second factory method do, when the ctor that exists right now doesn't work for it?